Full-text indexing fails due to missing Tika library

**Describe the bug**
When adding or updating an item in the search core (via the discovery consumer, e.g. after a metadata edit occurs in the UI), full-text indexing fails, with an error in dspace.log containing the following snippet:

```
Caused by: java.lang.NoClassDefFoundError: org/apache/tika/parser/csv/TextAndCSVParser
        at org.dspace.discovery.indexobject.IndexFactoryImpl.writeDocument(IndexFactoryImpl.java:99) ~[dspace-api-7.0-beta5-SNAPSHOT.jar:7.0-beta5-SNAPSHOT]
```

**To Reproduce**
Steps to reproduce the behavior:
1. Create an item with a PDF
2. Run filter-media on it so that a fulltext bitstream is created in the TEXT bundle
3. Edit the item to modify the metadata
4. Notice that the indexing fails and a message like the above appears in dspace.log

**Expected behavior**
Full-text indexing should succeed without error.

--- a/dspace-api/pom.xml
+++ b/dspace-api/pom.xml
@@ -632,7 +632,7 @@
         <!-- Used for full-text indexing with Solr -->
         <dependency>
             <groupId>org.apache.tika</groupId>
-            <artifactId>tika-core</artifactId>
+            <artifactId>tika-parsers</artifactId>
         </dependency>
         <dependency>
             <groupId>org.apache.lucene</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -1196,6 +1196,29 @@
                 <groupId>org.apache.tika</groupId>
                 <artifactId>tika-parsers</artifactId>
                 <version>1.24.1</version>
+                <!-- Exclude a few tika-parsers dependencies that we already use, with slightly different versions -->
+                <exclusions>
+                    <exclusion>
+                        <groupId>com.google.protobuf</groupId>
+                        <artifactId>protobuf-java</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>net.sf.ehcache</groupId>
+                        <artifactId>ehcache-core</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>jakarta.ws.rs</groupId>
+                        <artifactId>jakarta.ws.rs-api</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>javax.xml.soap</groupId>
+                        <artifactId>javax.xml.soap-api</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>org.jvnet.staxex</groupId>
+                        <artifactId>stax-ex</artifactId>
+                    </exclusion>
+                </exclusions>
             </dependency>
             <!-- Reminder: Keep icu4j (in Parent POM) synced with version used by lucene-analyzers-icu below,
              otherwise ICUFoldingFilterFactory may throw errors in tests.  -->

--- a/pom.xml
+++ b/pom.xml
@@ -1194,7 +1194,7 @@
             <!-- Used for full-text indexing with Solr. Should be synced with version of Tika in solr-cell -->
             <dependency>
                 <groupId>org.apache.tika</groupId>
-                <artifactId>tika-core</artifactId>
+                <artifactId>tika-parsers</artifactId>
                 <version>1.24.1</version>
             </dependency>
             <!-- Reminder: Keep icu4j (in Parent POM) synced with version used by lucene-analyzers-icu below,

